02 -  Front-end/Bloco-15-gerenciamento-de-estado-com-redux/dia-1-introducao-ao-redux-o-estado-global-da-aplicacao/exercices-1/index.js

```diff
@@ -0,0 +1,75 @@
+const ESTADO_INICIAL = {
+    colors: ['white', 'black', 'red', 'green', 'blue', 'yellow'],
+    index: 0,
+};
+
+const NEXT_COLOR = {
+    type: 'NEXT_COLOR',
+};
+
+const PREVIOUS_COLOR = {
+    type: 'PREVIOUS_COLOR',
+};
+
+const RADOM_COLOR = {
+    type: 'RADOM_COLOR',
+};
+
+function criarCor() {
+    const oneChar = ['1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'];
+    let cor = '#';
+    const aleatorio = () => Math.floor(Math.random() * oneChar.length);
+    for (let i = 0; i < 6; i += 1) {
+        cor += oneChar[aleatorio()];
+    }
+    return cor;
+}
+
+const reducer = (state = ESTADO_INICIAL, action) => {
+    switch(action.type) {
+        case 'NEXT_COLOR':
+            return {
+                ...state,
+                index: ((state.index > 4) ? 0 : state.index + 1),
+            };
+        case 'PREVIOUS_COLOR':
+            return {
+                ...state,
+                index: ((state.index < 1) ? 0 : state.index - 1),
+            };
+        case 'RADOM_COLOR':
+            return {
+                ...state,
+                colors: [...state.colors, criarCor()],
+                index: state.colors.length,
+            };
+        default:
+            return state
+    };
+};
+
+const store = Redux.createStore(reducer);
+
+store.subscribe(() => {
+    const { colors, index } = store.getState();
+    const div = document.querySelector('#container');
```

Comment:
Div ficou muito genérico 🤔, acho que algo mais parecido com isso ia ser melhor:
```suggestion
    const colorContainer = document.querySelector('#container');
```

Reply:
Verdade, vou fazer essa mudança! vlw